common/utilities/tests/test_timing.py

```diff
@@ -0,0 +1,207 @@
+
+        with self.subTest("Async version"):
+            res = await self.default_method_async()
+            log_mock.assert_called_with(5, 'default_method_async')
+            self.assertEqual(5, res)
+
+    @patch('common.utilities.timing.Stopwatch.stop_timer')
+    @patch('common.utilities.timing._log_time')
+    @async_test
+    async def test_exception_thrown_whilst_timing(self, log_mock, time_mock):
+        time_mock.return_value = 5
+        with self.subTest("Sync"):
+            try:
+                self.throw_error_method()
+            except ValueError:
+                pass
+            log_mock.assert_called_with(5, 'throw_error_method')
+
+        with self.subTest("Async"):
+            try:
+                await self.throw_error_method_async()
+            except ValueError:
+                pass
+
+            log_mock.assert_called_with(5, 'throw_error_method_async')
+
+    @patch('common.utilities.timing.Stopwatch.stop_timer')
+    @patch('logging.info')
+    @async_test
+    async def test_invoke_with_time_parameters(self, log_mock, time_mock):
+        with self.subTest("Sync"):
+            time_mock.return_value = 5
+            res = self.take_parameters("whew", 1, [2], {3: 3})
+            log_mock.assert_called_with('Method name=take_parameters took duration=5')
+            self.assertEqual("whew1", res)
+        with self.subTest("Async"):
+            time_mock.return_value = 5
+            res = await self.take_parameters_async("whew", 1, [2], {3: 3})
+            log_mock.assert_called_with('Method name=take_parameters_async took duration=5')
+            self.assertEqual("whew1", res)
+
+    @patch('common.utilities.timing.Stopwatch.stop_timer')
+    @patch('logging.info')
+    @async_test
+    async def test_async_times_execution_correctly(self, log_mock, time_mock):
+        time_mock.return_value = 0
+        task = self.default_method_async()
+
+        # check the method doesn't get timed until awaited
+        time_mock.return_value = 2
+
+        await task
+
+        log_mock.assert_called_with('Method name=default_method_async took duration=2')
+
+    @patch('common.utilities.timing.Stopwatch.stop_timer')
+    @patch('logging.info')
+    @async_test
+    async def test_invoke_with_time_varargs(self, log_mock, time_mock):
+        with self.subTest("Sync"):
+            time_mock.return_value = 5
+            res = self.var_parameters("whew", 1, 2, 3, 4, 5)
+            log_mock.assert_called_with('Method name=var_parameters took duration=5')
+            self.assertEqual("whew12345", res)
+        with self.subTest("Async"):
+            time_mock.return_value = 5
+            res = await self.var_parameters_async("whew", 1, "three", 4)
+            log_mock.assert_called_with('Method name=var_parameters_async took duration=5')
+            self.assertEqual("whew1three4", res)
+
+    @timing.time_function
+    def default_method(self):
+        return "slept"
```

Comment:
Why not also have this as:
```suggestion
        return 5
```

Reply:
Made them both into strings and made it a constant since it was used in a few places, this should also address your later comment stating it's a bit unclear having the response value matching the value I set the time as too